cmd/scollector/collectors/snmp_cisco.go

```diff
@@ -41,11 +35,59 @@ func SNMPCisco(cfg conf.SNMP) {
 		},
 	}
 
-	collectors = append(collectors, &IntervalCollector{
-		F: func() (opentsdb.MultiDataPoint, error) {
-			return GenericSnmp(cfg, mib)
+	collectors = append(collectors,
+		&IntervalCollector{
+			F: func() (opentsdb.MultiDataPoint, error) {
+				return GenericSnmp(cfg, mib)
+			},
+			Interval: time.Second * 30,
+			name:     fmt.Sprintf("snmp-cisco-%s", cfg.Host),
 		},
-		Interval: time.Second * 30,
-		name:     fmt.Sprintf("snmp-cisco-%s", cfg.Host),
-	})
+		&IntervalCollector{
```

Comment:
if you make the integrator here and pass it into c_cisco_cpu, you may not need the map or mutex at all.